lib/multi.c

```diff
@@ -3303,6 +3314,9 @@ static CURLMcode multi_socket(struct Curl_multi *multi,
   if(first)
     sigpipe_restore(&pipe_st);
 
+  if(run_conn_cache)
+    Curl_conncache_multi_perform(multi);
```

Comment:
I think this should be done *before* `sigpipe_restore()`, shouldn't it? I figure there is a risk that it would use TLS on a dead connection.

Reply:
Fixed in latest push.